docs/data-builder-dummy-data.md

```diff
@@ -0,0 +1,39 @@
+# Supplying a dummy dataset
+
+---8<-- 'includes/data-builder-danger-header.md'
+
+Because OpenSAFELY doesn't allow direct access to individual patient
+records, researchers must use a **dummy dataset** for developing their analytic
+code on their own computer.
+
+A dummy dataset needs to be similar enough to a real dataset to allow
+the analysis scripts to run successfully, but it must be randomly (or
+psuedo-randomly) generated and not contain data relating to real
+patients.
+
+In future, Data Builder will be able to generate suitable dummy data
+from a dataset definition, but for now you must generate your own dummy
+data file and use the `--dummy-data-file` argument to tell Data Builder
+where it is. When running on your own computer (or in the automated
+tests), Data Builder will copy this file to the same place it will put
+the real data when it is run inside the secure environment. This allows
+the rest of the analysis code to run in the same way whether it is
+working with real or dummy data.
+
+Data Builder will check that the column names and types of value in the
+dummy dataset match what will be in the real dataset and will raise an
+error if it finds a discrepancy.
+
+## Example
+
+```yaml title="Minimal Data Builder project YAML example"
+version: '3.0'
+
+actions:
+
+  generate_dataset:
+    run: databuilder:v0 generate_dataset --dataset-definition analysis/dataset_definition.py --dummy-data-file dummy_data.csv --output output/dataset.csv
```

Comment:
It seems a shame that the most important information is hidden. How about...

```suggestion
    run: >
      databuilder:v0 generate_dataset
        --dataset-definition analysis/dataset_definition.py
        --dummy-data-file dummy_data.csv
        --output output/dataset.csv
```

Reply:
OH YES! I didn't realise YAML supported this. Thank you! We should be using this everywhere.

Reply:
[The StackOverflow question about line breaks in YAML](https://stackoverflow.com/questions/3790454/how-do-i-break-a-string-in-yaml-over-multiple-lines) is blub studies at its finest.